File: leetcode/AddBinary/addBinary.java
public class Solution {
    public String addBinary(String a, String b) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int alen = a.length(), blen = b.length(), len = Math.max(alen, blen);
        StringBuilder sb = new StringBuilder();
        int cur = 0, next = 0;
        for (int i = 0; i < len; i++) {
            int abit = i < alen ? a.charAt(alen - 1 - i) - '0' : 0;
            int bbit = i < blen ? b.charAt(blen - 1 - i) - '0' : 0;
            cur = abit + bbit + next;
            next = cur / 2;
            sb.append(cur % 2);
        }
        if (next > 0) sb.append(1);
        return sb.reverse().toString();
    }
}

File: leetcode/AddBinary/addBinary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int la = a.size(), lb = b.size();
        int add = 0;
        string ans = "";
        for (int i = 0; i < max(la, lb); i++) {
            int ca = i < la ? a[la - 1 - i] - '0' : 0;
            int cb = i < lb ? b[lb - 1 - i] - '0' : 0;
            int res = ca + cb + add;
            ans = to_string(res % 2) + ans;
            add = res / 2;
        }
        if (add) {
            ans = "1" + ans;
        }
        return ans;
    }
};
